### ISSUE TITLE
could u provide simple code to extract features from a image?

### ISSUE THREAD

**Issue by MengHao666** [facebookresearch/dinov2#542]
I could not find the official insturctions on how to preprocess the original image for different dinov2 models.

**Comment by Antuke** [comment]
Hello, if you are intrested this is one way to do it:

`import torch
from PIL import Image
import sys 
import torch.nn.functional as F

REPO_PATH = "/mnt/c/Users/antonio/Desktop/dinov2" # here would be your path to your dinov2 repository
sys.path.append(REPO_PATH)

def load_image_from_url(url: str) -> Image:
    import urllib
    with urllib.request.urlopen(url) as f:
        return Image.open(f).convert("RGB")

from dinov2.hub.backbones import dinov2_vitl14 # in the file dinov2.hub there are differents model you can load
from dinov2.data.transforms import make_classification_eval_transform 

model = dinov2_vitl14().eval().to('cuda')
transform = make_classification_eval_transform()

img1_dog = load_image_from_url("https://upload.wikimedia.org/wikipedia/commons/9/93/Golden_Retriever_Carlos_%2810581910556%29.jpg")
img2_dog = load_image_from_url("https://upload.wikimedia.org/wikipedia/commons/f/fb/Welchcorgipembroke.JPG")
img3_cat = load_image_from_url("https://upload.wikimedia.org/wikipedia/commons/4/4d/Cat_November_2010-1a.jpg")



img1_prep = transform(img1_dog).unsqueeze(0).to('cuda')
img2_prep = transform(img2_dog).unsqueeze(0).to('cuda')
img3_prep = transform(img3_cat).unsqueeze(0).to('cuda')

with torch.no_grad():
    feat1 = model(img1_prep) # [1, 384]
    feat2 = model(img2_prep) # [1, 384]
    feat3 = model(img3_prep) # [1, 384]

    # If you want to take the patc features
    # feat1_p = model.get_intermediate_layers(img1_prep)[0] # [1, 256, 384]
    # feat2_p = model.get_intermediate_layers(img2_prep)[0] # [1, 256, 384]
    # feat3_p = model.get_intermediate_layers(img2_prep)[0] # [1, 256, 384]

feat1_norm = F.normalize(feat1, dim=1)
feat2_norm = F.normalize(feat2, dim=1)
feat3_norm = F.normalize(feat3, dim=1)

cos_sim_dog1_dog2 = F.cosine_similarity(feat1_norm, feat2_norm, dim=1)
cos_sim_dog1_cat = F.cosine_similarity(feat1_norm, feat3_norm, dim=1)
cos_sim_dog2_cat = F.cosine_similarity(feat2_norm, feat3_norm, dim=1)

print(f'Sim between dog1 & dog2 = {cos_sim_dog1_dog2}')
print(f'Sim between dog1 & cat = {cos_sim_dog1_cat}')
print(f'Sim between dog2 & cat = {cos_sim_dog2_cat}')`
